iModelCore/ECDb/Tests/NonPublished/SchemaGraphTests.cpp

```diff
@@ -0,0 +1,532 @@
+    locater.AddSchemaString(eKey, eXml);
+    context->AddSchemaLocater(locater);
+    SanitizingSchemaLocater sanitizingLocater(m_ecdb.GetSchemaLocater());
+    context->SetFinalSchemaLocater(sanitizingLocater);
+
+    ECSchemaPtr a = context->LocateSchema(aKey, SchemaMatchType::LatestReadCompatible);
+    ASSERT_TRUE(a.IsValid());
+    ECSchemaPtr d = context->LocateSchema(dKey, SchemaMatchType::LatestReadCompatible);
+    ASSERT_TRUE(d.IsValid()); // locating d out of order before b and c
+    ECSchemaPtr b = context->LocateSchema(bKey, SchemaMatchType::LatestReadCompatible);
+    ASSERT_TRUE(b.IsValid());
+    ECSchemaPtr c = context->LocateSchema(cKey, SchemaMatchType::LatestReadCompatible);
+    ASSERT_TRUE(c.IsValid());
+    ECSchemaPtr e = context->LocateSchema(eKey, SchemaMatchType::LatestReadCompatible);
+    ASSERT_TRUE(e.IsValid());
+
+    // import schemas and verify their versions
+    SchemaImportResult result = m_ecdb.Schemas().ImportSchemas({ a.get(), c.get(), d.get(), e.get() });
+    ASSERT_EQ(SchemaImportResult::OK, result);
+    m_ecdb.SaveChanges();
+
+    // Assert the stored versions inside ECDb
+    AssertECDbSchemaVersion(aKey);
+    AssertECDbSchemaVersion(bKey);
+    AssertECDbSchemaVersion(cKey);
+    AssertECDbSchemaVersion(dKey);
+    AssertECDbSchemaVersion(eKey);
+
+    // Assert the referened schemas in the tree
+    auto* aFromE = GetReferencedSchema("A", *e);
+    ASSERT_TRUE(aFromE != nullptr);
+    ASSERT_TRUE(aFromE->GetSchemaKey().Matches(aKey, SchemaMatchType::Exact));
+
+    auto* bFromE = GetReferencedSchema("B", *e);
+    ASSERT_TRUE(bFromE != nullptr);
+    ASSERT_TRUE(bFromE->GetSchemaKey().Matches(bKey, SchemaMatchType::Exact));
+
+    auto* cFromE = GetReferencedSchema("C", *e);
+    ASSERT_TRUE(cFromE != nullptr);
+    ASSERT_TRUE(cFromE->GetSchemaKey().Matches(cKey, SchemaMatchType::Exact));
+
+    auto* dFromE = GetReferencedSchema("D", *e);
+    ASSERT_TRUE(dFromE != nullptr);
+    ASSERT_TRUE(dFromE->GetSchemaKey().Matches(dKey, SchemaMatchType::Exact));
+
+    auto* cFromD = GetReferencedSchema("C", *d);
+    ASSERT_TRUE(cFromD != nullptr);
+    ASSERT_TRUE(cFromD->GetSchemaKey().Matches(cKey, SchemaMatchType::Exact));
+
+    auto* bFromC = GetReferencedSchema("B", *c);
+    ASSERT_TRUE(bFromC != nullptr);
+    ASSERT_TRUE(bFromC->GetSchemaKey().Matches(bKey, SchemaMatchType::Exact));
+
+    auto* aFromB = GetReferencedSchema("A", *b);
+    ASSERT_TRUE(aFromB != nullptr);
+    ASSERT_TRUE(aFromB->GetSchemaKey().Matches(aKey, SchemaMatchType::Exact));
+
+    // Confirm in-memory references are the same
+    ASSERT_TRUE(a.get() == aFromE);
+    ASSERT_TRUE(b.get() == bFromE);
+    ASSERT_TRUE(c.get() == cFromE);
+    ASSERT_TRUE(d.get() == dFromE);
+    ASSERT_TRUE(c.get() == cFromD);
+    ASSERT_TRUE(b.get() == bFromC);
+    ASSERT_TRUE(a.get() == aFromB);
+    }
+    }
+
+#define ENABLE_IMPORT_SCHEMAS_FROM_FILES_TEST 0
+
+#if ENABLE_IMPORT_SCHEMAS_FROM_FILES_TEST
```

Comment:
this type of test should not be checked in.  It is useless without the original schema files which will not be checked in.

Reply:
removed. I was going to reuse this for future defects, but I can keep a local copy instead.